base/channels.jl

```diff
@@ -105,7 +105,7 @@ true
   The following constructors were added in Julia 1.3.
 
 Other constructors:
-* `Channel{T}(func::Function, sz=0)`
+* `Channel{T}(func::Function, size)`
 * `Channel{T}(func::Function; csize=0, taskref=nothing)`
```

Comment:
Is it too late to change these keyword APIs? They're pretty bad. `csize` would be much better spelled `size`—yes, it's the "channel size" but what else would it be the size of? It's the `Channel` constructor. The `taskref` argument is an ok name, but man, this is a really strange API. Is there anywhere else we use a keyword argument to let someone pass in a `Ref` which we populate with an optional output?

Reply:
> Is it too late to change these keyword APIs? They're pretty bad.

Yeah I completely agree. I was under the impression that we can't change the API for these because of the compatibility promise for julia 1.0? I guess we could at least deprecate the old interface, maybe? Or am I not understanding how constrained we are here? If we can change them, i'm definitely on board!

I think Channels are going to be an important part of our cooperative multithreading story, so it'd be nice to get this API right.

Reply:
And actually, I want to also open another PR that adds maybe a `spawn=` kwarg that when `true` creates the new Task for the provided function as a multithreading task with `sticky=false`. Something like `Channel(spawn=true) do ch ; ... end`?

I was planning on opening a separate PR for that, just wanted to leave this as a heads-up, while we're thinking about this.

Reply:
Totally agree. `csize` and `ctype` should go. `Channel(ctype = Int)` should obviously be `Channel{Int}()`. `taskref` is weird but at least is solves a real problem (we want the constructor to return a channel, but you might want to get the task as well). The `spawn` keyword arg would be OK too.

Reply:
Could the task just be accessible as a field of the Channel object?

Reply:
> Totally agree. `csize` and `ctype` should go.

So this is something we can still do? Backwards compatibility isn't a concern? +1 to removing those

Reply:
We should do a quick grep over all packages to gauge how often `csize` and `ctype` have been used. But `ctype` in particular is so non-idiomatic that I'd be comfortable removing it.

Reply:
We can also define multiple keywords and note in the docs that some will be deprecated in the next release. There was a fair few of those in 0.7

Reply:
> We should do a quick grep over all packages to gauge how often `csize` and `ctype` have been used.

There seems to be 23 uses of `csize` across 9 packages + julia base:
https://pkg.julialang.org/docs/search?q=%5Cbcsize%5Cs%2a%3D

And 99 uses of `ctype` across 34 packages + julia base (if i counted that right):
https://pkg.julialang.org/docs/search?q=%5Cbctype%5Cs%2a%3D

Reply:
From Triage call:
 - change to the new API, remove the old API from the docs, and describe it as "deprecated" in the News.
 - And maybe start sending PRs to packages to switch to the new API?

Reply:
Okay this is done. (The changes to julia, not sending PRs to packages)